src/gpu/nvidia/cudnn_convolution.cpp

```diff
@@ -47,24 +47,21 @@ status_t cudnn_convolution_fwd_t::execute_convolution(
         const bool use_temp_dst = pd()->use_temp_dst();
 
         if (with_scratchpad) {
-            scratch_acc = std::make_shared<scratch_acc_t>(
-                    utils::downcast<sycl::sycl_buffer_memory_storage_t *>(
-                            ctx.get_scratchpad_grantor()
-                                    .get_memory_storage(memory_tracking::names::
-                                                    key_conv_cudnn_algo)
-                                    .get())
-                            ->buffer()
-                            .get_access<::sycl::access::mode::read_write>(cgh));
+            scratch_acc = std::make_shared<scratch_acc_t>(CTX_SCRATCH_ACCESSOR(
```

Comment:
Hi! I feel the knowledge discovered deserves a separate comment here (or below next to another scratchpad access) so that it's clear what is going on and why offset is needed.

Another thought then came to me is to have another macro which would take care of adding offset automatically when pushing memory into arguments. Could you explore that option, please? Tracking such places by hands is not the best solution, at least to me. Thank you.